### PR TITLE
Fix ES module for Node.js

### PR DESCRIPTION
**Pull request by denysoblohin-okta**
Issue: 
Can't import `broadcast-channel` as [ES module for Node.js](https://nodejs.org/docs/latest/api/esm.html).
```
import { BroadcastChannel } from 'broadcast-channel';
         ^^^^^^^^^^^^^^^^
SyntaxError: Named export 'BroadcastChannel' not found. The requested module 'broadcast-channel' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'broadcast-channel';
const { BroadcastChannel } = pkg;

    at ModuleJob._instantiate (internal/modules/esm/module_job.js:124:21)
    at async ModuleJob.run (internal/modules/esm/module_job.js:179:5)
    at async Loader.import (internal/modules/esm/loader.js:178:24)
    at async Object.loadESM (internal/process/esm_loader.js:68:5)
    at async handleMainPromise (internal/modules/run_main.js:59:12)
```
Steps to reproduce: create simple project with `index.mjs` (having just 1 line with import of broadcast-channel) and try to run `node index.mjs`

Solution:
- Use `.mjs` extensions for ESM module. (Don't add `type: "module" in package.json` because we want to support both ESM and CJS)
- Add testing ESM and CJS modules

More details here: https://github.com/sheremet-va/dual-packaging

### CODE REVIEW

**Comment by pubkey** [comment]
Please do not check in generated files from the dist folder. This makes the PR unreadable.

**Comment by denysoblohin-okta** [comment]
@pubkey Done

File: package.json [hunk 1]
```diff
@@ -101,6 +104,7 @@
     "@types/core-js": "2.5.5",
     "assert": "2.0.0",
     "async-test-util": "2.0.0",
+    "babel-plugin-replace-import-extension": "^1.1.3",
```
Comment (pubkey) [inline package.json:107]: Please pin the version, no `^`.
I do not use a package-lock in libs for several reasons, so versions must be pinned.

**Comment by pubkey** [comment]
Hmm, the CI fails.
But it also fails on master.
Looks like a timing problem where github actions now gives a slower server then before.

**Comment by pubkey** [comment]
Yes please.

**Comment by denysoblohin-okta** [comment]
@pubkey [Done](https://github.com/pubkey/broadcast-channel/pull/962/commits/12e647926ac556620722cf7d0daccfa27bb12528). [CI passes](https://github.com/denysoblohin-okta/broadcast-channel/actions/runs/2380641519). Please approve workflow for PR.

**Comment by pubkey** [comment]
Merged and released.
Thank you.

**Pull request merged**